botocore/args.py

```diff
@@ -641,14 +642,16 @@ def _build_endpoint_resolver(
             client_endpoint_url=endpoint_url,
             legacy_endpoint_url=endpoint.host,
         )
-        # botocore does not support client context parameters generically
-        # for every service. Instead, the s3 config section entries are
-        # available as client context parameters. In the future, endpoint
-        # rulesets of services other than s3/s3control may require client
-        # context parameters.
-        client_context = (
-            s3_config_raw if self._is_s3_service(service_name_raw) else {}
-        )
+        # Client context params for s3 conflict with the available settings
+        # in the `s3` parameter on the `Config` object. The s3 config will
+        # always take precedence over the client context params for s3 and
+        # s3control if set.
+        if self._is_s3_service(service_name_raw) and s3_config_raw:
+            client_context = s3_config_raw
```

Comment:
You've set up the documentation to exclude old params, but are documenting any new ones. If I supply my old config but want to opt into a new client context param, what does that look like? It's not clear how this handles that resolution.

Reply:
Yeah you're right. I'll take another look at this.